### ISSUE TITLE
Get the bot to detect Nextbots

### ISSUE THREAD

**Issue by T-Rizzle12** [T-Rizzle12/Custom-Gmod-Bot#12]

**Comment by strubium** [comment]
Can the bot read the `ai_relationship` values of nextbots? Perhaps you could make it if a nextbot "Hates" the player it will attack the nextbot. 

It also could be that most nextbot creators don't give their creations hitboxes or health, especially the .png ones that are made with free software and aren't made by hand.

**Comment by T-Rizzle12** [comment]
> Can the bot read the `ai_relationship` values of nextbots? Perhaps you could make it if a nextbot "Hates" the player it will attack the nextbot.
> 
> It also could be that most nextbot creators don't give their creations hitboxes or health, especially the .png ones that are made with free software and aren't made by hand.

Nope, Nextbots use different ways to mark enemies and so far I have no idea on how to grab it. Hardcoding looks likes the only method so far. I still want to find a way to make this dynamic.

**Comment by strubium** [comment]
> > Can the bot read the `ai_relationship` values of nextbots? Perhaps you could make it if a nextbot "Hates" the player it will attack the nextbot.
> > It also could be that most nextbot creators don't give their creations hitboxes or health, especially the .png ones that are made with free software and aren't made by hand.
> 
> Nope, Nextbots use different ways to mark enemies and so far I have no idea on how to grab it. Hardcoding looks like the only method so far. I still want to find a way to make this dynamic.

Dang it! Yeah, hardcoding seems to be the only option right now.  The [valve dev community](https://developer.valvesoftware.com/wiki/Example_of_NextBot_Behaviors) talks about how TF2 nextbots move about through a game, but they just use a crap ton of hardcoded actions.

**Comment by T-Rizzle12** [comment]
> > > Can the bot read the `ai_relationship` values of nextbots? Perhaps you could make it if a nextbot "Hates" the player it will attack the nextbot.
> > > It also could be that most nextbot creators don't give their creations hitboxes or health, especially the .png ones that are made with free software and aren't made by hand.
> > 
> > 
> > Nope, Nextbots use different ways to mark enemies and so far I have no idea on how to grab it. Hardcoding looks like the only method so far. I still want to find a way to make this dynamic.
> 
> Dang it! Yeah, hardcoding seems to be the only option right now. The [valve dev community](https://developer.valvesoftware.com/wiki/Example_of_NextBot_Behaviors) talks about how TF2 nextbots move about through a game, but they just use a crap ton of hardcoded actions.

I have been thinking about it and I might have a way to partially automate it. Most Nextbot addons have a way to check if a Nextbot is controlled by it. It will take a bit of hardcoding but I can make this work. As long as the Nextbot works on a base, I can use that base to check NPC relationships and automate how my bot checks for enemies and allies.

**Comment by strubium** [comment]
Nice! Good idea, I would not think of that.
